code/modules/mob/living/blood.dm

```diff
@@ -320,20 +319,27 @@
 				drop.transfer_mob_blood_dna(src)
 				return
 			else
-				temp_blood_DNA = drop.return_blood_DNA() //we transfer the dna from the drip to the splatter
-				qdel(drop)//the drip is replaced by a bigger splatter
+				return
+	else
+		var/obj/effect/decal/cleanable/blood/splatter/split = locate() in T
+		if(split)
+			if(split.splats < 10)
+				split.splats++
+				var/mutable_appearance/blood = new /mutable_appearance()
+				blood.icon = split.icon
+				blood.icon_state = pick(split.random_icon_states)
+				blood.pixel_x = rand(-8,8)
+				blood.pixel_y = rand(-8,8)
+				split.add_overlay(blood)
+				split.transfer_mob_blood_dna(src)
+				return
+			else
+				return
 		else
-			drop = new(T, get_static_viruses())
-			drop.transfer_mob_blood_dna(src)
+			split = new(T, get_static_viruses())
+			split.transfer_mob_blood_dna(src)
 			return
-
-	new /obj/effect/decal/cleanable/blood/splatter(T, get_static_viruses())
-	var/obj/effect/decal/cleanable/blood/B = locate() in T
 	//MOJAVE SUN EDIT END - Blood Sprites
-	B.bloodiness = min((B.bloodiness + BLOOD_AMOUNT_PER_DECAL), BLOOD_POOL_MAX)
-	B.transfer_mob_blood_dna(src) //give blood info to the blood decal.
-	if(temp_blood_DNA)
-		B.add_blood_DNA(temp_blood_DNA)
```

Comment:
Move split splat code down to line 331 (compare with pre blood-change code)
If B is found, do your split splat overlay code
Else create splatter as normal

revert your removal of bloodiness - it is used for footprints
keep DNA, unless there is a reason to remove it

otherwise looks good - Mutable appearance is a good idea, great way to handle your pixel shifts,  really clever

Reply:
Done, should be good to go.